tangelo/linq/target/target_stim.py

```diff
@@ -0,0 +1,95 @@
+# Copyright 2023 Good Chemistry Company.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""The stim simulation package is made for fast simulation of Clifford circuits.
+See https://arxiv.org/abs/2103.02202v3 for more details.
+"""
+
+from collections import Counter
+
+import numpy as np
+
+from tangelo.linq import Circuit
+from tangelo.linq.target.backend import Backend
+from tangelo.linq.translator import translate_circuit as translate_c
+from tangelo.linq.translator.translate_stim import translate_tableau
+from tangelo.linq.helpers.circuits.measurement_basis import pauli_of_to_string
+
+
+class StimSimulator(Backend):
+    """Interface to the stim simulator"""
+    def __init__(self, n_shots=None, noise_model=None):
+        import stim
+        super().__init__(n_shots=n_shots, noise_model=noise_model)
+        self.stim = stim
+
+    def simulate_circuit(self, source_circuit: Circuit, return_statevector=False, initial_statevector=None):
+        """Perform state preparation corresponding to the input circuit on the
+        target backend, return the frequencies of the different observables, and
+        either the statevector or None depending on the availability of the
+        statevector and if return_statevector is set to True. For the
```

Comment:
It's probably an old copy paste from the Backend class or something, but here we know exactly what this class can do or not do, and we should cut off whatever options and parameters are not available, and tell things the way they are.